File: 3620-maximum-number-of-distinct-elements-after-operations/3620-maximum-number-of-distinct-elements-after-operations.cpp
class Solution {
public:
    int maxDistinctElements(vector<int>& nums, int k) {
        int n = nums.size();
        sort(nums.begin(), nums.end());
        int ans = 0;
        int prev = -1e9;

        for(int i = 0; i<n; i++){
            int left_ele = max(nums[i]-k, prev+1);
            if(left_ele<=nums[i]+k){
                prev = left_ele;
                ans++;
            }
        }
        return ans;
    }
};

File: 3620-maximum-number-of-distinct-elements-after-operations/3620-maximum-number-of-distinct-elements-after-operations.py
class Solution(object):
    def maxDistinctElements(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """

        nums.sort()
        ans, prev = 0, -10**9
        for i in nums:
            left_ele = max(i-k, prev+1)
            if(left_ele <= i+k):
                prev = left_ele
                ans += 1
        return ans        
        